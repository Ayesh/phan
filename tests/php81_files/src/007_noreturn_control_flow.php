<?php

class Example7 {
    public static function custom_throws(string $message): never {
        throw new RuntimeException(__METHOD__ . ": $message");
    }
}
function custom_throws(string $message): never {
    throw new RuntimeException($message);
}
if (!isset($x) || !is_array($x)) {
    custom_throws('message');
}

echo spl_object_hash($x); // should warn, this is an array
Example7::custom_throws('finished');

echo "Unreachable\n";
