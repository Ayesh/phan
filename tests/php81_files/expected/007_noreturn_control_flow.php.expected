%s:15 PhanTypeMismatchArgumentInternalReal Argument 1 ($object) is $x of type array but \spl_object_hash() takes object
%s:18 PhanPluginUnreachableCode Unreachable statement detected
